### ISSUE TITLE
Wrong error indication when using Self

### ISSUE THREAD

**Issue by amotzop** [microsoft/pyright#3710]
## Environment data
-   Language Server version: 2022.7.30
-   OS and version: Ubuntu 20.04
-   Python version: 3.9.5

## Code Snippet
```python
from typing_extensions import Self

class Base:
    def func(self) -> Self:
        return self

class Derived(Base):
    def func(self) -> Self:
        return super().func()
```
Pylance indicates the last line with an error, `Expression of type "Base" cannot be assigned to return type "Self@Derived"`

## Expected behavior
There shouldn't be an error, the return type of `super().func()` should be `Derived`

## Actual behavior

There is an error and wrong return type indications

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.261, which I just published. It will also be included in an upcoming release of pylance.

**Comment by amotzop** [comment]
I just checked it with Pylance versions 2022.7.41 and 2022.7.42 (which both have pyright version at least 1.1.261), and this bug still happens to me.

**Comment by erictraut** [comment]
Reopening the issue. It looks like I may have fixed part of the problem, but it may still exist on another code path.

**Comment by erictraut** [comment]
This will be fixed in the next release.

As I suspected, there was another code path that needed to be fixed. This is a tricky case to test because the bug occurs only when expressions are evaluated in a certain order. Of course, the behavior should be the same regardless of evaluation order. But it's difficult to test all potential orderings.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.265, which I just published. It will be included in a future release of pylance.
